pytransform3d/test/test_transform_manager.py

```diff
@@ -521,3 +583,106 @@ def test_numpy_timeseries_transform_wrong_input_shapes():
+    # create entities A and B together with their transformations from world
+    duration = 10.0  # [s]
+    sample_period = 0.5  # [s]
+    velocity_x = 1  # [m/s]
+    time_A, pq_arr_A = create_sinusoidal_movement(
+        duration, sample_period, velocity_x, y_start_offset=0.0, start_time=0.1
+    )
+    A2world = NumpyTimeseriesTransform(time_A, pq_arr_A)
+
+    time_B, pq_arr_B = create_sinusoidal_movement(
+        duration, sample_period, velocity_x, y_start_offset=2.0,
+        start_time=0.35
+    )
+    B2world = NumpyTimeseriesTransform(time_B, pq_arr_B)
+
+    tm = TemporalTransformManager()
+
+    tm.add_transform("A", "W", A2world)
+    tm.add_transform("B", "W", B2world)
+
+    query_times = np.array([4.9, 5.2])  # [s]
+    A2B_at_query_time = tm.get_transform_at_time("A", "B", query_times)
+
+    origin_of_A_pos = pt.vector_to_point([0, 0, 0])
+    origin_of_A_in_B_pos1 = pt.transform(A2B_at_query_time[0], origin_of_A_pos)
+    origin_of_A_in_B_xyz1 = origin_of_A_in_B_pos1[:-1]
+
+    origin_of_A_in_B_x1, origin_of_A_in_B_y1 = \
+        origin_of_A_in_B_xyz1[0], origin_of_A_in_B_xyz1[1]
+
+    assert origin_of_A_in_B_x1 == pytest.approx(-1.11, abs=1e-2)
+    assert origin_of_A_in_B_y1 == pytest.approx(-1.28, abs=1e-2)
+
+
+def test_temporal_transform_manager_incorrect_frame():
+    duration = 10.0  # [s]
+    sample_period = 0.5  # [s]
+    velocity_x = 1  # [m/s]
+    time_A, pq_arr_A = create_sinusoidal_movement(
+        duration, sample_period, velocity_x, y_start_offset=0.0, start_time=0.1
+    )
+    A2world = NumpyTimeseriesTransform(time_A, pq_arr_A)
+
+    tm = TemporalTransformManager()
+    tm.add_transform("A", "W", A2world)
+    with pytest.raises(KeyError, match="Unknown frame"):
+        tm.get_transform("B", "W")
+    with pytest.raises(KeyError, match="Unknown frame"):
+        tm.get_transform("A", "B")
+
+    tm = TemporalTransformManager(check=False)
+    tm.add_transform("A", "W", A2world)
+    with pytest.raises(ValueError):
+        tm.get_transform("B", "W")
+    with pytest.raises(ValueError):
+        tm.get_transform("A", "B")
+
+
+def test_temporal_transform_manager_out_of_bounds():
+    duration = 10.0  # [s]
+    sample_period = 0.5  # [s]
+    velocity_x = 1  # [m/s]
+    time_A, pq_arr_A = create_sinusoidal_movement(
+        duration, sample_period, velocity_x, y_start_offset=0.0, start_time=0.0
+    )
+    A2world = NumpyTimeseriesTransform(time_A, pq_arr_A, time_clipping=True)
+
+    time_B, pq_arr_B = create_sinusoidal_movement(
+        duration, sample_period, velocity_x, y_start_offset=2.0, start_time=0.1
+    )
+    B2world = NumpyTimeseriesTransform(time_B, pq_arr_B, time_clipping=True)
+
+    tm = TemporalTransformManager()
+    tm.add_transform("A", "W", A2world)
+    tm.add_transform("B", "W", B2world)
+
+    assert min(time_A) == 0.0
+    assert min(time_B) == 0.1
+    A2B_at_start_time, A2B_before_start_time = tm.get_transform_at_time(
```

Comment:
@JonasHablitzel This is a great example of how readable the new vectorized version of the interface is. :)